domain/modelagent/state/state.go

```diff
@@ -75,3 +75,54 @@ WHERE uuid = $M.model_id
 	}
 	return vers, nil
 }
+
+// AgentVersionForModelName returns the agent version for the model with the
+// given name and owner.
+func (st *State) AgentVersionForModelName(
+	ctx context.Context,
+	username string,
+	modelName string,
+) (version.Number, error) {
+	db, err := st.DB()
+	if err != nil {
+		return version.Zero, errors.Trace(err)
+	}
+
+	q := `
+SELECT &dbAgentVersion.target_agent_version
+FROM v_model
+WHERE name = $M.model_name
+AND owner_name = $M.owner
+`
+
+	rval := dbAgentVersion{}
+	args := sqlair.M{
```

Comment:
Instead of using sqlair.M, we prefer to define a concrete struct.

Reply:
can't quite get this to work

Reply:
Is that the case for args? I thought this rule only applied on the way out @ycliuhw ?

Reply:
done now